src/classes/User/PassageUser.ts

```diff
@@ -0,0 +1,290 @@
+        try {
+            const response = await this.usersApi.updateUser({
+                userId,
+                appId: this.config.appId,
+                updateUserRequest: options,
+            });
+
+            return response.user;
+        } catch (err) {
+            if (err instanceof ResponseError) {
+                throw await PassageError.fromResponseError(err, 'Could not update user');
+            }
+
+            throw err;
+        }
+    }
+
+    /**
+     * Create a user.
+     *
+     * @param {CreateUserRequest} args Arguments for creating a user
+     * @return {Promise<PassageUser>} Passage User object
+     */
+    public async create(args: CreateUserRequest): Promise<PassageUser> {
+        try {
+            const response = await this.usersApi.createUser({
+                appId: this.config.appId,
+                createUserRequest: args,
+            });
+
+            return response.user;
+        } catch (err) {
+            if (err instanceof ResponseError) {
+                throw await PassageError.fromResponseError(err, 'Could not create user');
+            }
+
+            throw err;
+        }
+    }
+
+    /**
+     * Delete a user using their user ID.
+     *
+     * @param {string} userId The userID used to delete the corresponding user.
+     * Either an E164 phone number or email address.
+     * @return {boolean} True if user was deleted, false if not
+     */
+    public async delete(userId: string): Promise<boolean> {
+        try {
+            await this.usersApi.deleteUser({
+                userId,
+                appId: this.config.appId,
+            });
+            return true;
+        } catch (err) {
+            if (err instanceof ResponseError) {
+                throw await PassageError.fromResponseError(err, 'Could not delete user');
+            }
+
+            throw err;
+        }
+    }
+
+    /**
+     * Get a user's devices using their user ID.
+     *
+     * @param {string} userId The Passage user ID
+     * @return {Promise<WebAuthnDevices[]>} List of devices
+     */
+    public async listDevices(userId: string): Promise<WebAuthnDevices[]> {
+        try {
+            const response = await this.userDevicesApi.listUserDevices({
+                userId,
+                appId: this.config.appId,
+            });
+
+            return response.devices;
+        } catch (err) {
+            if (err instanceof ResponseError) {
+                throw await PassageError.fromResponseError(err, "Could not fetch user's devices:");
+            }
+
+            throw err;
+        }
+    }
+
+    /**
+     * Revoke a user's device using their user ID and the device ID.
+     *
+     * @param {string} userId The Passage user ID
+     * @param {string} deviceId The Passage user's device ID
+     * @return {Promise<boolean>}
+     */
+    public async revokeDevice(userId: string, deviceId: string): Promise<boolean> {
```

Comment:
nitpick: what do you think of deprecating this and using a single object param instead to match the other function signatures that have multiple inputs to enable invocation via destructuring?

Reply:
Yeah I'm down for that for consistency's sake (plus end users will have named params which is nice).

Reply:
oh right I remember why I didn't do this in this PR - To deprecate it I'd have to give the new function a new name (Javascript doesn't support function overloading much to my endless frustration)